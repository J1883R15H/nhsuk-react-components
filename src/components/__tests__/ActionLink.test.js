import React from 'react';
import { shallow, mount } from 'enzyme';
import ActionLink from '../action-link';

describe('ActionLink', () => {
  it('matches the snapshot', () => {
    const shallowMount = shallow(<ActionLink>Test</ActionLink>);
    expect(shallowMount).toMatchSnapshot();
    shallowMount.unmount();
  });
  it('sets props on openInNewTab', () => {
    const currentWindow = mount(<ActionLink>Test</ActionLink>);
    const newTab = mount(<ActionLink openInNewWindow>Test</ActionLink>);
    const currentWindowProps = currentWindow
      .find('.nhsuk-action-link__link')
      .props();
    const newTabProps = newTab.find('.nhsuk-action-link__link').props();

    expect(newTabProps.rel).toBe('noreferrer noopener');
    expect(newTabProps.target).toBe('_blank');
    expect(currentWindowProps.rel).toBe('');
    expect(currentWindowProps.target).toBe('');
  });
});
