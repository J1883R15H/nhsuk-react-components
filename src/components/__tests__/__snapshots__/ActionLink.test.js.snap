// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionLink matches the snapshot 1`] = `
<div
  className="nhsuk-action-link"
  style={Object {}}
>
  <a
    className="nhsuk-action-link__link"
  >
    <ArrowRightCircle
      className=""
      height={32}
      width={32}
    />
    <span
      className="nhsuk-action-link__text"
    >
      Test
    </span>
  </a>
</div>
`;
